ctest-next/templates/test.rs

```diff
@@ -82,7 +83,430 @@ mod generated_tests {
             for (i, (&b1, &b2)) in ptr1_bytes.iter().zip(ptr2_bytes.iter()).enumerate() {
                 // HACK: This may read uninitialized data! We do this because
                 // there isn't a good way to recursively iterate all fields.
-                check_same_hex(b1, b2, &format!("{{ ident }} value at byte {}", i));
+                check_same_hex(b1, b2, &format!("{{ ident }} value at byte {i}"));
+            }
+        }
+    }
+    {%- endif +%}
+    {%- endfor +%}
+
+    {%- for alias in ffi_items.aliases() +%}
+    {%- let ident = alias.ident() +%}
+
+    /// Test that the size and alignment of the aliased type is the same in both Rust and C.
+    /// 
+    /// This can fail if a different type is used on one side, and uses the built in size and
+    /// alignment functions to check.
+    pub fn size_align_{{ ident }}() {
+        extern "C" {
+            fn __test_size_{{ ident }}() -> u64;
+            fn __test_align_{{ ident }}() -> u64;
+        }
+        unsafe {
+            check_same(mem::size_of::<{{ ident }}>() as u64,
+                __test_size_{{ ident }}(), "{{ ident }} size");
+            check_same(mem::align_of::<{{ ident }}>() as u64,
+                __test_align_{{ ident }}(), "{{ ident }} align");
+        }
+    }
+
+    {%- if translator.has_sign(ffi_items, alias.ty) +%}
+
+    /// Test that the aliased type has the same sign (signed or unsigned) in both Rust and C.
```

Comment:
"sign"->"signedness" is the word for describing whether or not a type is signed.